go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -327,12 +378,34 @@ func (oc *Controller) addEgressFirewallRules(hashedAddressSetNameIPv4, hashedAdd
 				matchTargets = []matchTarget{{matchKindV4CIDR, rule.to.cidrSelector}}
 			}
 		} else {
-			// rule based on DNS NAME
-			dnsNameAddressSets, err := oc.egressFirewallDNS.Add(ef.namespace, rule.to.dnsName)
+			var addressSet addressset.AddressSet
```

Comment:
This is now a very complicated function - it's hard to follow. Can you think of a way to clean it up? Maybe separate some concerns a bit?